tests/Microsoft.Identity.Test.Integration/SeleniumTests/InteractiveFlowTests.cs

```diff
@@ -76,12 +80,15 @@ public async Task Interactive_AADAsync()
         }
 
         [TestMethod]
+        [TestCategory("AzureUSGovernment")]
```

Comment:
So if we want to run ALL tests (i.e. all 4 clouds), I would have to run 4 test runs with 4 different config files. 

I would propose something else - have 4 tests (one for each cloud), each decorated as here with [TestCategory("cloud")]. If we want to run all tests, this is the default. If we want to run just cloud1 and cloud2, you easilty do that via mstest by filtering by test category. This way you don't need a config file. 

Next what's going to happen is that we'll have too many individual tests and it will get tedious to write each of them. You can either use Mark's tehnique with DataRows or generate the signatures with T4 (I can show you how).

Reply:
I decided to go with data rows. They add lots of flexibility that makes it easy for us to run tests in other clouds without manipulating a config file and with minimal copied code. Now we can simply point to the cloud category in the mstest task to run all cloud tests